subworkflows/local/bam_impute_glimpse1/main.nf

```diff
@@ -19,15 +19,33 @@ workflow BAM_IMPUTE_GLIMPSE1 {
     ch_versions = Channel.empty()
     ch_multiqc_files = Channel.empty()
 
+        // Channels for branching
+    ch_input = ch_input
+        .branch {
+            bam: it[1] =~ 'bam|cram'
+            vcf: it[1] =~ '(vcf|bcf)(.gz)*'
+            other: true
+        }
+    ch_input.other
+        .map{ error "Input files must be either BAM/CRAM or VCF/BCF" }
+
     // Glimpse1 subworkflow
     BAM_GL_BCFTOOLS( // Compute GL for input data once per panel by chromosome
-        ch_input,
+        ch_input.bam,
         ch_posfile,
         ch_fasta
     )
     ch_multiqc_files = ch_multiqc_files.mix(BAM_GL_BCFTOOLS.out.multiqc_files)
     ch_versions = ch_versions.mix(BAM_GL_BCFTOOLS.out.versions)
 
+    // Combine input and chunks reference
+    ch_impute = ch_input.vcf
+        .combine(ch_posfile)
+        .map{ metaI, vcf, index, metaPC, legend ->
+            [metaI + ["panel": metaPC.id, "chr": metaPC.chr], vcf, index]
+        }
+        .mix(BAM_GL_BCFTOOLS.out.vcf_tbi)
```

Comment:
Do we have a test case for this one? Do we need documentation for this in `usage.md`? In which cases would users want to use `vcf`? Examples would be nice to have :)

Reply:
Glimpse1 can also be used to impute vcf files, this could be used to impute snp chip array later or also allow user to use their vcf if they already have called the variants.

Reply:
I'll add more informations in the usage.md.
We could also add a test case in the subworkflows.